### DISCUSSION TITLE
Homekit doorbell ring notification period

### DISCUSSION THREAD

**Discussion by barisonal** [HomeSpan/HomeSpan#537] [General]
I have just explored that there is a notification limit for doorbell rings in Homekit. I tested and looks like devices can receive a "Doorbell rang" notification per minute. 

I don't know is it Apple Homekit policy or parametric but I couldn't find any information indicates the policy.

I have also found @HomeSpan 's this [answer](https://github.com/HomeSpan/HomeSpan/discussions/34#discussioncomment-356764) in another discussion. As far as I understand there is no any limit in Homespan.

Has anyone ever experienced this?

**Comment by HomeSpan** [comment]
I was testing this out on my own setup and did not encounter any limit. I was able to cause a chime on my HomePod every second by repeated presses of a button. 

The rule for providing updates is only a recommendation.  For example, don't send an update for temperature every second. 

But a doorbell ring is and event and there is no reason to delay sending in HomeSpan.  

If HomeKit does not respond, I'm not sure why.

**Reply by barisonal** [reply]
> I was testing this out on my own setup and did not encounter any limit. I was able to cause a chime on my HomePod every second by repeated presses of a button.
> 
> The rule for providing updates is only a recommendation. For example, don't send an update for temperature every second.
> 
> But a doorbell ring is and event and there is no reason to delay sending in HomeSpan.
> 
> If HomeKit does not respond, I'm not sure why.

I understood the problem. Yes you are right, there is no any limit.

My `button` function was like below;

```
void button(int pin, int pressType) override {
 
   switchEvent->setVal(pressType);                 
      
  }
```

So it was sending 0 for each press and on that case there was a limit on Homekit notification because of I send same value (0) for each time. I explored that it sends notification if I set different values for each time then I updated button function by using a flag named `lastCalledState` and it sends the value by switching between 1 and 0.

```
void button(int pin, int pressType) override {
    
    lastCalledState = !lastCalledState;
    switchEvent->setVal(lastCalledState);                 
      
  }
```

**Reply by ronny-antoon** [reply]
So you solved the problem.!?

**Reply by barisonal** [reply]
Yes!

**Comment by ronny-antoon** [comment]
> I have just explored that there is a notification limit for doorbell rings in Homekit. I tested and looks like devices can receive a "Doorbell rang" notification per minute. 
> 
> 
> 
> I don't know is it Apple Homekit policy or parametric but I couldn't find any information indicates the policy.
> 
> 
> 
> I have also found @HomeSpan 's this [answer](https://github.com/HomeSpan/HomeSpan/discussions/34#discussioncomment-356764) in another discussion. As far as I understand there is no any limit in Homespan.
> 
> 
> 
> Has anyone ever experienced this?

Do you want to send notification from doorbell more!?

**Reply by barisonal** [reply]
Actually no, but 1 minute is not an enough gap between two different bells. So if someone miss the first call there might be second time which means second chance :)
